tests/repo/mod.rs

```diff
@@ -52,7 +55,36 @@ pub fn base_dir_filter(entry: &DirEntry) -> bool {
     }
 }
 
-pub fn clone_rust() {
-    let result = Command::new("tests/clone.sh").status().unwrap();
-    assert!(result.success());
+pub fn clone_rust() -> io::Result<()> {
+    if match read_to_string("tests/rust/COMMIT") {
+        Err(_) => true,
+        Ok(contents) => contents.trim() != REVISION,
+    } {
+        let url = format!("https://github.com/rust-lang/rust/archive/{}.zip", REVISION);
+        let mut tmpfile = tempfile::tempfile().unwrap();
```

Comment:
Would it be possible to do this without saving the file to disk? These files are reasonably small.

Reply:
I will check on that.

Reply:
I haven't found a way other than memmap, which I'm not comfortable adding that much complexity.